### PR TITLE
Do not show diagnostic warning for disconnected iOS devices

### LINKED ISSUE #105970: `flutter doctor`/`devices` shows disconnected devices "iPhone is not connected. Xcode will continue when iPhone is connected. (code -13)"
These errors can show up when the device was set up for wireless pairing.  It has also been reported for other disconnected devices that were never set up with wireless pairing but I'm not sure why that happens.
```
[✓] Connected device (3 available)
    • iPhone 11 (mobile) • CB23A1C8-7131-457E-9B14-6721BC996C3D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64     • macOS 12.4 21F79 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 102.0.5005.61
    ! Error: iPhone is not connected. Xcode will continue when iPhone is connected. (code -13)
    ! Error: iPad is not connected. Xcode will continue when iPad is connected. (code -13)
```
Filter it out.

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/105970

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/macos/xcode_test.dart [hunk 1]
```diff
@@ -788,6 +788,35 @@ void main() {
...
+    "simulator" : false,
+    "modelName" : "iPad Pro (12.9-inch) (3rd generation)",
+    "error" : {
+      "code" : -13,
+      "failureReason" : "",
+      "underlyingErrors" : [
+        {
+          "code" : 4,
+          "failureReason" : "",
+          "description" : "iPad is locked.",
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/macos/xcode_test.dart:803]: Is this really what the JSON looks like? it says both that it's locked and not connected? that's very confusing...

File: packages/flutter_tools/test/general.shard/macos/xcode_test.dart [hunk 2]
```diff
@@ -799,6 +828,7 @@ void main() {
 
           final List<String> errors = await xcdevice.getDiagnostics();
           expect(errors, hasLength(4));
+          // Does not contain "Xcode will continue when iPad is connected."
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/macos/xcode_test.dart:831]: Can you write an expectation for this? Maybe something like:

`expect(errors, isNot(contains('Xcode will continue when iPad is connected.')));`

File: packages/flutter_tools/test/general.shard/macos/xcode_test.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/test/general.shard/macos/xcode_test.dart:803]: Yes that's exactly copied from my results.  It's not connected, it claims it's a usb interface, but it somehow knows it's locked. But it doesn't actually know, when I unlock the device it doesn't try to wireless pair or something.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**